ldap/servers/slapd/back-ldbm/db-mdb/mdb_instance.c

```diff
@@ -374,8 +374,17 @@ dbmdb_open_all_files(dbmdb_ctx_t *ctx, backend *be)
     int i;
 
     if (!ctx) {
-        struct ldbminfo *li = (struct ldbminfo *)(be->be_database->plg_private);
-        ctx = MDB_CONFIG(li);
+        if (!be) {
+            /* Testing for "be" to avoid a covscan warning although 
+              * dbmdb_open_all_files is never called with both parameters NULL
+              */
+            slapi_log_err(SLAPI_LOG_ERR, "dbmdb_open_all_files",
+                          "Unable to open the database environment witout either the database context or a backend.\n");
+            return DBI_RC_INVALID;
```

Comment:
Or you may use the famous dbmdb_map_error(__FUNCTION_, DBI_RC_INVALID) to log the backtrace

Reply:
Indeed, but IMHO the backtrace is useless and I prefer avoiding using a sledgehammer to break a nut! :wink: 
The only case to trigger the failure is either that memory get corrupted or that we added new broken code
(So far the only case dbmdb_open_all_files is called with a NULL backend is at startup when setting up the database environment context and in that case the context that was just created is provided.
In the other case (typically when a backend is (re)started the backend is provided)